### PR TITLE
vm/agent_groups

### PR DESCRIPTION
**Pull request by milanaghera-tenable**
This PR contains following changes :

Agent Groups API related changes

- removed list method and replace it with search method
- marshmallow schema validation for request body
- Agent Bulk Operation related changes

Agent Bulk Operation API related changes

- added support for below endpoint as part of agent_groups.py module

SDK method -> Corresponding endpoint
send_instruction_to_agents_in_group -> POST api/v3/agent-groups/{group_id}/agents/_bulk/directive

### CODE REVIEW

**Review by ngaglani-tenable — APPROVED** [review]

**Review by ngaglani-tenable — APPROVED** [review]

File: tests/io/v3/vm/agent_groups/test_agent_groups_schema.py [hunk 1]
```diff
@@ -1,41 +1,68 @@
 '''
-Testing the AgentGroups schemas
+Test cases for AgentGroups schemas
 '''
 from tenable.io.v3.vm.agent_groups.schema import AgentGroupSchema
```
Comment (alpesh-te) [inline tests/io/v3/vm/agent_groups/test_agent_groups_schema.py]: There should be name consistency across the module `AgentGroupSchema` should be `AgentGroupsSchema`

File: tenable/io/v3/vm/agent_groups/schema.py [hunk 2]
```diff
@@ -1,7 +1,40 @@
...
+    Supportive schema for AgentSchema
+    '''
+    type = fields.Str(
+        validate=validate.OneOf(['restart', 'settings']),
+        required=True
+    )
+    option = fields.Dict()
+
+
+class AgentGroupFilterSchema(ParseFilterSchema):
```
Comment (alpesh-te) [inline tenable/io/v3/vm/agent_groups/schema.py:33]: There should be name consistency across the module  `AgentGroupFilterSchema` should be `AgentGroupsFilterSchema`

File: tests/io/v3/vm/agent_groups/test_agent_groups_schema.py [hunk 3]
```diff
@@ -1,41 +1,68 @@
...
-            '57b74c0a-5d95-11ec-bf63-0242ac130002',
-            '57b74e58-5d95-11ec-bf63-0242ac130002',
-            '57b74f66-5d95-11ec-bf63-0242ac130002'
-        ]
+    assert payload == schema.dump(schema.load(payload))
+
+    # Test for schema validation for details method
+    payload = {
+        'filters': [('platform', 'match', ['window'])],
+        'filter_type': 'and',
```
Comment (alpesh-te) [inline tests/io/v3/vm/agent_groups/test_agent_groups_schema.py:44]: Can you please give me more context about `filter_type` as payload? I couldn't find here https://confluence.eng.tenable.com/pages/viewpage.action?spaceKey=PORG&title=%5BSDK%5D+Agent+Groups

File: tests/io/v3/vm/agent_groups/test_agent_groups_schema.py [hunk 4]
```diff
@@ -1,41 +1,68 @@
...
+        'sort': [('name', 'asc'), ('platform', 'asc')],
+        'offset': 0,
+        'filters': [
+            {
+                'value': ['window'],
+                'operator': 'match',
+                'field': 'platform'
+            }
+        ],
+        'wildcard': 'IYKKQIGOBWXUXOZIBURFENPNMGZOSWBUKVCD',
```
Comment (alpesh-te) [inline tests/io/v3/vm/agent_groups/test_agent_groups_schema.py:61]: Can you please give me more context about `wildcard` as payload? I couldn't find here https://confluence.eng.tenable.com/pages/viewpage.action?spaceKey=PORG&title=%5BSDK%5D+Agent+Groups

File: tests/io/v3/vm/agent_groups/test_agent_groups_schema.py [hunk 5]
```diff
@@ -1,41 +1,68 @@
...
+        'limit': 50,
+        'filter_type': 'and',
+        'wildcard_fields': ['name']
```
Comment (alpesh-te) [inline tests/io/v3/vm/agent_groups/test_agent_groups_schema.py:64]: Can you please give me more context about `wildcard_fields` as payload? I couldn't find here https://confluence.eng.tenable.com/pages/viewpage.action?spaceKey=PORG&title=%5BSDK%5D+Agent+Groups

File: tests/io/v3/vm/agent_groups/test_agent_groups_api.py [hunk 6]
```diff
@@ -233,16 +229,256 @@ def test_task_status(api):
...
+        'created',
+        'modified',
+        'owner_id',
+        'agents_count'
+    ]
+
+    filter = {
+        'and': [{
+            'property': 'modified',
+            'operator': 'date-eq',
```
Comment (alpesh-te) [inline tests/io/v3/vm/agent_groups/test_agent_groups_api.py:376]: `date-eq` operator doesn't match with API Specification doc https://confluence.eng.tenable.com/display/UW/UW+API+Specification

**Review by alpesh-te — CHANGES_REQUESTED** [review]
I have given few comments

File: tests/io/v3/vm/agent_groups/test_agent_groups_api.py [hunk 7]
```diff
@@ -233,16 +229,256 @@ def test_task_status(api):
...
+
+    iterator = api.v3.vm.agent_groups.search(
+        fields=fields, filter=filter, sort=sort, limit=200
+    )
+    assert isinstance(iterator, SearchIterator)
+
+    exclusions_list = []
+    for item in iterator:
+        exclusions_list.append(item)
+    assert len(exclusions_list) == test_response['pagination']['total']
```
Comment (alpesh-te) [inline tests/io/v3/vm/agent_groups/test_agent_groups_api.py]: Refactor above code to get rid of loop

File: tests/io/v3/vm/agent_groups/test_agent_groups_schema.py (hunk 3 above)
Comment (milanaghera-tenable) [reply tests/io/v3/vm/agent_groups/test_agent_groups_schema.py:44]: @alpesh-te filter_type is a part of the `send_instruction_to_agents_in_group` method's request body and this particular method is related to agent bulk operation API also based on @tomeara 's suggestion we have included this endpoint in the agent_groups module. for more details please refer to the [agent bulk operation confluence page](https://confluence.eng.tenable.com/display/PORG/%5BSDK%5D+Agent+Bulk+Operation).

File: tests/io/v3/vm/agent_groups/test_agent_groups_schema.py (hunk 4 above)
Comment (milanaghera-tenable) [reply tests/io/v3/vm/agent_groups/test_agent_groups_schema.py:61]: @alpesh-te `wildcard `is a part of the `send_instruction_to_agents_in_group `method's request body. for more details please refer to the [agent bulk operation confluence page](https://confluence.eng.tenable.com/display/PORG/%5BSDK%5D+Agent+Bulk+Operation).

File: tests/io/v3/vm/agent_groups/test_agent_groups_schema.py (hunk 5 above)
Comment (milanaghera-tenable) [reply tests/io/v3/vm/agent_groups/test_agent_groups_schema.py:64]: @alpesh-te `wildcard_fields` is a part of the `send_instruction_to_agents_in_group` method's request body. for more details please refer to the [agent bulk operation confluence page](https://confluence.eng.tenable.com/display/PORG/%5BSDK%5D+Agent+Bulk+Operation).

File: tests/io/v3/vm/agent_groups/test_agent_groups_schema.py (hunk 1 above)
Comment (milanaghera-tenable) [reply tests/io/v3/vm/agent_groups/test_agent_groups_schema.py]: @alpesh-te As per our discussion on call, will keep the schema name as it is.

File: tenable/io/v3/vm/agent_groups/schema.py (hunk 2 above)
Comment (milanaghera-tenable) [reply tenable/io/v3/vm/agent_groups/schema.py:33]: @alpesh-te similar to AgentGroupSchema, will keep the AgentGroupFilterSchema name as it is.

File: tests/io/v3/vm/agent_groups/test_agent_groups_api.py (hunk 6 above)
Comment (milanaghera-tenable) [reply tests/io/v3/vm/agent_groups/test_agent_groups_api.py:376]: pre-existing operators were used as filters in this endpoint and more information can be found in ref link : https://jira.eng.tenable.com/browse/IN-582?focusedCommentId=1528634&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1528634

**Review by alpesh-te — APPROVED** [review]
LGTM!

**Pull request merged**